addons/account/partner.py

```diff
@@ -241,11 +241,32 @@ def _debit_search(self, cr, uid, obj, name, args, context=None):
     def _invoice_total(self, cr, uid, ids, field_name, arg, context=None):
         result = {}
         account_invoice_report = self.pool.get('account.invoice.report')
-        for partner in self.browse(cr, uid, ids, context=context):
-            domain = [('partner_id', 'child_of', partner.id), ('state', 'not in', ['draft', 'cancel'])]
+        user = self.pool['res.users'].browse(cr, uid, uid, context=context)
+        for partner_id in ids:
+            # first read invoices in the user's currency
+            all_partner_ids = self.pool['res.partner'].search(
+                cr, uid, [('id', 'child_of', partner_id)], context=context)
```

Comment:
Either use a context with active_test=False or test if all_partner_ids is empty to prevent
```
ProgrammingError: syntax error at or near ")"
LINE 1: ... from account_invoice_report where partner_id in () and stat...
```
when setting a partner without underlying contacts to inactive.